bundles/framework/myplaces3/ButtonHandler.js

```diff
@@ -0,0 +1,417 @@
+                            drawMode: 'area',
+                            shape: "Polygon",
+                            drawOptions: drawOpt
+                        });
+                    };
+                    measureTool.iconCls = 'tool-measure-area';
+                    measureTool.tooltip = me.loc('tools.measurearea.tooltip');
+                    sandbox.request(me, reqBuilder(tool, me.measureButtonGroup, measureTool));
+                }
+            }; */
+
+            for (tool in this.buttons) {
+                if (!this.buttons.hasOwnProperty(tool)) {
+                    continue;
+                }
+                sandbox.request(this, reqBuilder(tool, this.buttonGroup, this.buttons[tool]));
+                
+                // for logged-in-user: add line & area measurement tools if configured
+                // basic measurement tools should be configured off so these can be used
+                // as replacements - this way measurement can be saved as myplaces
+                // TODO: this is not the way to do this: instead allow request to be used for
+                // saving a place and make the basic measurement tools use it when available
+                // TODO uncomment this if you need to add AdditionalMeasureTools
+                /*
+                if (Oskari.user().isLoggedIn() && addAdditionalMeasureTools) {
+                    if (tool === 'line') {
+                        addMeasureTool(tool);
+                    }
+                    if (tool === 'area') {
+                        addMeasureTool(tool);
+                    }
+                }*/
+            }
+
+            var user = Oskari.user();
+            if (!user.isLoggedIn()) {
+                // disable toolbar buttons for guests
+                this.disableButtons();
+            } else {
+                // logged in user -> listen to events as normal
+                for (p in me.eventHandlers) {
+                    
+                    if (me.eventHandlers.hasOwnProperty(p)) {
+                        sandbox.registerForEventByName(me, p);
+                    }
+                }
+            }
+        },
+        /**
+         * @method disableButtons
+         * Disables draw buttons
+         */
+        disableButtons: function () {
+            var sandbox = this.instance.sandbox,
+                stateReqBuilder = sandbox.getRequestBuilder('Toolbar.ToolButtonStateRequest');
+            sandbox.request(this, stateReqBuilder(undefined, this.buttonGroup, false));
+        },
+        /**
+         * @method startNewDrawing
+         * Sends a StartDrawingRequest with given params. Changes the panel controls to match the application state (new/edit)
+         * @param config params for StartDrawingRequest
+         */
+        sendDrawRequest: function (config) {
+            this.drawMode = config.drawMode;
+            this.instance.setIsFinishedDrawing(false);
+            this.instance.enableGfi(false);
```

Comment:
DrawTools does the enabling/disabling gfi internally now. Is this required anymore?